mycore-base/src/main/java/org/mycore/datamodel/metadata/MCRMetaPersonName.java

```diff
@@ -301,10 +268,22 @@ public String getAffix() {
      * This method set the affix text element.
      */
     public void setAffix(String affix) {
-        if (affix != null) {
-            this.affix = affix.trim();
-        } else {
-            this.affix = "";
+        this.affix = affix != null ? affix.trim() : "";
+    }
+
+    private void updateName() {
```

Comment:
I think the whole update method is unnecessary
I would move the checks for empty and the specified calculated fallbacks into the corresponding getters
getFirstname(), getCallName(), getFullName();

``` java
public String getCallname(){
  callname = callname.trim();
  return callname.isEmpty() ? callname : firstname.trim();
}
```
Otherwise you can not distinguish between explicitely set values and calculated defaults.

Reply:
I removed the update method.